Sensory/src/main/java/com/google/android/sensory/fhir_data/ViewHolderFactoryUtil.kt

```diff
@@ -45,5 +54,42 @@ class ViewHolderFactoryUtil {
       itemView.findViewById<View>(com.google.android.fhir.datacapture.R.id.error).visibility =
         View.GONE
     }
+
+    // Function to rotate the image bitmap based on its Exif information
+    fun rotateImageIfRequired(context: Context, bitmap: Bitmap, uri: Uri): Bitmap {
+      return try {
+        val exif = context.contentResolver.openInputStream(uri)?.let { ExifInterface(it) }
+
+        val rotationInDegrees = when (exif?.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)) {
+          ExifInterface.ORIENTATION_ROTATE_90 -> 90
+          ExifInterface.ORIENTATION_ROTATE_180 -> 180
+          ExifInterface.ORIENTATION_ROTATE_270 -> 270
+          else -> 90
+        }
+
+        val matrix = Matrix()
+        matrix.postRotate(rotationInDegrees.toFloat())
+        Bitmap.createBitmap(bitmap, 0, 0, bitmap.width, bitmap.height, matrix, true)
+
+      } catch (e: IOException) {
+        e.printStackTrace()
+        bitmap
+      }
+    }
+
+    class RotateTransformation(private val context: Context, private val uri: Uri?) : BitmapTransformation() {
+      override fun transform(
+        pool: BitmapPool,
+        toTransform: Bitmap,
+        outWidth: Int,
+        outHeight: Int
+      ): Bitmap {
+        return rotateImageIfRequired(context, toTransform, uri!!)
+      }
+
+      override fun updateDiskCacheKey(messageDigest: MessageDigest) {
```

Comment:
Since we are not caching the image, do we need to override this ?
If this is not mandatory to override, then we can skip it.